File: eecs168/2021summer/2021.07.22.operator-overloading/Triangle.cpp
//Triangle.cpp

#include "Triangle.h"
/*
Triangle::Triangle()
{
	//Initialize your member variables
	base = 0;
	height = 0;
}
*/
Triangle::Triangle(double newBase, double newHeight)
{
	//Where's the object? 
	//We're in the object!
	setBase(newBase);
	setHeight(newHeight);
}

double Triangle::getBase() const 
{
	return(base);
}

double Triangle::getHeight() const
{
	return(height);	
}

bool Triangle::setBase(double newBase)
{
	if(newBase > 0)
	{
		base = newBase;
		return(true);
	}
	else
	{
		base = 0;
		return(false);
	}
}

bool Triangle::setHeight(double newHeight)
{
	if(newHeight > 0)
	{
		height = newHeight;
		return(true);
	}
	else
	{
		this->height = 0;
		return (false);
	}
}

double Triangle::area() const
{
	return(0.5 * this->base * this->height);
	return(0.5 * base * height);
}

bool Triangle::operator>(const Triangle& rhs) const
{
	return(this->area() > rhs.area());
}

bool Triangle::operator<=(const Triangle& rhs) const
{
	return(!((*this) > rhs));//uses our existing definition
}

File: eecs168/2025spring/MWF/2025.04.25.classes-part-6/triangle/triangle.py
#triangle.py
class Triangle:
    def __init__(self, base, height):
        print("Triangle's init called")
        #member variables
        self.base = base
        self.height = height

    def area(self):
        ans = 0.5*self.base*self.height
        return ans

    def __gt__(self, other):
        print('__gt__ called')
        if self.area() > other.area():
            return True
        else:
            return False

    def __str__(self):
        output_str = 'Triangle with base = '
        output_str += str(self.base)
        output_str += ' and height = '
        output_str += str(self.height)
        return output_str

    def __repr__(self):
        output_str = f'Triangle({self.base}, {self.height})'
        return output_str
